sql/core/src/main/scala/org/apache/spark/sql/execution/ExistingRDD.scala

```diff
@@ -101,10 +102,12 @@ private[sql] case class PhysicalRDD(
     override val metadata: Map[String, String] = Map.empty,
     isUnsafeRow: Boolean = false,
     override val outputPartitioning: Partitioning = UnknownPartitioning(0))
-  extends LeafNode {
+  extends LeafNode with CodegenSupport {
 
   protected override def doExecute(): RDD[InternalRow] = {
-    if (isUnsafeRow) {
+    val conf = SQLContext.getActive().get
+    // The vectorized reader does not produce UnsafeRows. In this case we will convert.
+    if (isUnsafeRow && !conf.getConf(SQLConf.PARQUET_VECTORIZED_READER_ENABLED)) {
```

Comment:
We should also check that the data source is Parquet or not. I'd suggest to do this in `createFromDataSource`